sdk/nodejs/automation/stack.ts

```diff
@@ -564,6 +564,95 @@ Event: ${line}\n${e.toString()}`);
         };
     }
 
+    /**
+     * Import resources into the stack
+     *
+     * @param options Options to specify resources and customize the behavior of the import.
+     */
+    async import(options: ImportOptions): Promise<ImportResult> {
+        const args = ["import", "--yes", "--skip-preview"];
+
+        if (options.message) {
+            args.push("--message", options.message);
+        }
+
+        applyGlobalOpts(options, args);
+
+        let importResult: CommandResult;
+        // for import operations, generate a temporary directory to store the following:
+        //   - the import file when the user specifies resources to import
+        //   - the output file for the generated code
+        // we use the import file as input to the import command
+        // we the output file to read the generated code and return it to the user
+        let tempDir: string = "";
+        try {
+            tempDir = await fs.promises.mkdtemp(pathlib.join(os.tmpdir(), "pulumi-import-"));
+            const tempGeneratedCodeFile = pathlib.join(tempDir, "generated-code.txt");
+            if (options.resources) {
+                // user has specified resources to import, write them to a temp import file
+                const tempImportFile = pathlib.join(tempDir, "import.json");
+                await fs.promises.writeFile(
+                    tempImportFile,
+                    JSON.stringify({
+                        nameTable: options.nameTable,
+                        resources: options.resources,
+                    }),
+                );
+
+                args.push("--file", tempImportFile);
+            }
+
+            if (options.generateCode === false) {
+                // --generate-code is set to true by default
+                // only use --generate-code=false if the user explicitly sets it to false
+                args.push("--generate-code=false");
+            } else {
+                args.push("--out", tempGeneratedCodeFile);
+            }
+
+            if (options.protect === false) {
+                // --protect is set to true by default
+                // only use --protect=false if the user explicitly sets it to false
+                args.push(`--protect=false`);
+            }
+
+            if (options.converter) {
+                // if the user specifies a converter, pass it to `--from <converter>` argument of import
+                args.push("--from", options.converter);
+                if (options.converterArgs) {
+                    // pass any additional arguments to the converter
+                    // for example using {
+                    //   converter: "terraform"
+                    //   converterArgs: "./tfstate.json"
+                    // }
+                    // would be equivalent to `pulumi import --from terraform ./tfstate.json`
+                    args.push(...options.converterArgs);
```

Comment:
Same comment as the dotnet PR, there should probably be a "--" arg here to split the arg list

Reply:
Done ✅